### ISSUE TITLE
Unable to cast Mystical Dispute on Mystical Dispute for U

### ISSUE THREAD

**Issue by Terminus42** [magefree/mage#6684]
I was unable to cast a Mystical Dispute on my opponent's Mystical Dispute targeting my blue spell with one mana open for it. My opponent was at 3 mana and paid 1 to cast his.

**Comment by Terminus42** [comment]
The blue spell in question was Thopter Foundry. I was producing the blue mana to cast my Mystical Dispute with a Steam Vents.

**Comment by Terminus42** [comment]
No T3feris or other cards that could possibly have interacted with the situation were involved.

**Comment by Terminus42** [comment]
The same exact thing seems to happen every time when you try to counter an Ice-Fang Coatl with a Mystical Dispute with less than three mana available, possibly when you try to counter any multicolored blue spell.

**Comment by Terminus42** [comment]
Other people on XMage.de are reporting the same issue.

**Comment by LevelX2** [comment]
https://github.com/magefree/mage/blob/ce5e61e612bf65fd5cec6c4c75b63435fe5f087a/Mage.Sets/src/mage/cards/m/MysticalDispute.java#L31

Zone has to be All instead of Stack.

**Comment by LevelX2** [comment]
The same Problem concerns the following cards:
```
BoltBend.java
MysticalDispute.java
GustOfWind.java
IgneousElemental.java
IntoTheStory.java
LookoutsDispersal.java
OfOneMind.java
PriceOfFame.java
OakhameAdversary.java
SavageStomp.java
NotOfThisWorld.java
SirenReaver.java
ScourTheLaboratory.java
StormwingEntity.java
StoicRebuttal.java
WingedWords.java
WizardsLightning.java
TitanicBrawl.java
WizardsRetort.java
```

**Comment by JayDi85** [comment]
Not all cards can be fixed by `Zone.STACK` changes. Some spells have special filters like "if you target creature type then do cost reduction" -- that thing must be brainstorming. 

As example -- introduce new method like:
* `boolean canActivateWithoutAvailableMana(ManaOptions availableMana)`

Some cards can calcs and checks possible targets and mana, but another can return true (e.g. allow user to put spell/ability on stack without available mana cheks). Also it can be cost reduction for other spells with same things (like all spells that target dino must cost less).

**Comment by JayDi85** [comment]
I updated verify test to find potential problem cards. Open `VerifyCardDataTest.java`, find 6684 text (that issue number) and uncommend two lines with warning and error. After that run `checkMissingCardData`. It will output problem cards to check.

Current output: 
https://pastebin.com/raw/kMTQSYV6

**Comment by SergeWind** [comment]
i feel like the reason people are complaining about this (and similar stuff like Hogaak, et al.) is because now you can't even _attempt_ to move spells onto stack if they aren't highlighted as "playable", when in the previous release you could. easy solution would be to revert whatever stops players from trying to play cards that the client thinks are not playable, hard solution would be to upgrade the code that determines playability to include all the cost reduction and alternate payment modes. but i'm not a coder, just wanted to drop my two cents on this issue, so i'll leave the hard decisions for you

**Comment by Zerrisx** [comment]
That sounds right to me.  There are a *lot* of ways to cast a card that might be hard to calculate.  For example, if you want to use Selvala, Explorer Returned without worrying about opponents getting priority due to effects triggering off the card draw, it is the strategically correct play to put a sometimes as-yet unaffordable spell on the stack.  A player shouldn't be hard locked out of that option.

**Comment by JayDi85** [comment]
@SergeWind 
It's already have special "switch" to enabled old logic in code for all cards:
![shot_200622_061639](https://user-images.githubusercontent.com/8344157/85242482-3b177d80-b450-11ea-8069-90a4533d99df.png)

But that's NOT a solution. All that not working cards were broken many years -- that cards can't be tested by unit tests, AI can't play that cards too. So it's a good time to collect feedback and find all problem to fix it.

So the real solution's steps:
1. Find all broken cards;
2. Try to fix it;
3. For single cards enable workaround (that can't be fixed by normal way);

**Comment by Zerrisx** [comment]
In that case, let's create a master issue (it could be this one) for this and start adding every card that gets reported to it, while closing all the individual card issues as duplicates.  #6680, #6685, #6696, and the reports in discord that you can't cast spells with an untapped Chrome Mox in play were probably all issues revealed by this.

**Comment by JayDi85** [comment]
@Zerrisx yes, it's will be useful to create new master issue like you do it before. Can you make it?

**Comment by Zerrisx** [comment]
> 
> 
> @Zerrisx yes, it's will be useful to create new master issue like you do it before. Can you make it?

Done - #6698

**Comment by dmbrtd** [comment]
> 
> 
> That sounds right to me. There are a _lot_ of ways to cast a card that might be hard to calculate. For example, if you want to use Selvala, Explorer Returned without worrying about opponents getting priority due to effects triggering off the card draw, it is the strategically correct play to put a sometimes as-yet unaffordable spell on the stack. A player shouldn't be hard locked out of that option.

I agree with Zerrisx. Even after all of these issues are addressed there should still be an escape hatch, maybe via right-click, to put an uncastable spell on the stack.
